mne/stats/regression.py

```diff
@@ -310,6 +311,9 @@ def solver(X, Y):
 
         # this is the magical part (thanks to Marijn van Vliet):
         # use toeplitz to construct series of diagonals
+        # XXX this is magical but terribly inefficient as
+        # linalg.toeplitz(samples, lags) allocates a dense matrix with
+        # n_samples rows for each condition in the loop !!!
         X[:, cum:cum + tmax_ - tmin_] = linalg.toeplitz(samples, lags)
```

Comment:
I think it might be possible to use [scipy.sparse.spdiags](http://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.sparse.spdiags.html)